src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -169,7 +169,8 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
       frameState.attributions, vectorSource.getAttributions());
   this.updateLogos(frameState, vectorSource);
 
-  if (!this.dirty_ && (frameState.viewHints[ol.ViewHint.ANIMATING] ||
+  if (!this.dirty_ && (!vectorLayer.getUpdateWhileAnimating() &&
+      frameState.viewHints[ol.ViewHint.ANIMATING] ||
       frameState.viewHints[ol.ViewHint.INTERACTING])) {
```

Comment:
If `updateWhileAnimating` is `true` (`!updateWhileAnimating` is `false`), `animating` is `false` and `interacting` is `true` the expression will evaluate to `true` (because `(!true && false || true)` is `true`). I think this is not correct. If `updateWhileAnimating` is `true` we want the expression to evaluate to `false` so it looks to me that the expression should be:

```
if (!this.dirty_ && !vectorLayer.getUpdateWhileAnimating() &&
    (frameState.viewHints[ol.ViewHint.ANIMATING] || frameState.viewHints[ol.ViewHint.INTERACTING])) {
  ...
}
```

Am I wrong?